connectd/connectd.c

```diff
@@ -259,6 +252,14 @@ static struct io_plan *peer_reconnected(struct io_conn *conn,
 
 	status_peer_debug(id, "reconnect");
 
+	/* If we have a previous reconnection, we replace it. */
+	pr = peer_reconnected_htable_get(&daemon->reconnected, id);
+	if (pr) {
+		status_broken("Reconnected AGAIN");
+		peer_reconnected_htable_del(&daemon->reconnected, pr);
```

Comment:
If the previous reconnection already had the destructor installed, would it not delete this twice?  Or delete from `htable` is idempotent?